swap/src/bitcoin/wallet.rs

```diff
@@ -295,6 +295,39 @@ impl<B, D, C> Wallet<B, D, C> {
     }
 }
 
+impl Wallet<(), bdk::database::MemoryDatabase, ()> {
+    /// Creates a new, funded wallet to be used within tests.
+    #[cfg(test)]
```

Comment:
I find it a bit hard to spot that this code is only used in the tests. I find the abstractions that you introduced for the wallet interesting and the resulting test code great. Just a bit hard to understand what overall functionality the wallet offers and what functionality is for tests only.
Do you see a possibility to extract test code somehow?

Reply:
I've moved the `test` block to the whole `impl` block, is that better?